dd-sdk-android/src/main/java/datadog/trace/common/processor/TraceProcessor.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package datadog.trace.common.processor;
+
+import datadog.opentracing.DDSpan;
+import datadog.trace.api.Config;
+import datadog.trace.common.processor.rule.AnalyticsSampleRateRule;
+import datadog.trace.common.processor.rule.DBStatementRule;
+import datadog.trace.common.processor.rule.ErrorRule;
+import datadog.trace.common.processor.rule.HttpStatusErrorRule;
+import datadog.trace.common.processor.rule.ResourceNameRule;
+import datadog.trace.common.processor.rule.SpanTypeRule;
+import datadog.trace.common.processor.rule.Status404Rule;
+import datadog.trace.common.processor.rule.URLAsResourceNameRule;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class TraceProcessor {
+  final Rule[] DEFAULT_RULES =
+      new Rule[] {
+        // Rules are applied in order.
+        new DBStatementRule(),
+        new ResourceNameRule(),
+        new SpanTypeRule(),
+        new ErrorRule(),
+        new HttpStatusErrorRule(),
+        new URLAsResourceNameRule(),
+        new Status404Rule(),
+        new AnalyticsSampleRateRule(),
+      };
+
+  private final List<Rule> rules;
+
+  public TraceProcessor() {
+
+    rules = new ArrayList<>(DEFAULT_RULES.length);
+    for (final Rule rule : DEFAULT_RULES) {
+      if (isEnabled(rule)) {
+        rules.add(rule);
+      }
+    }
+  }
+
+  private static boolean isEnabled(final Rule rule) {
+    boolean enabled = Config.get().isRuleEnabled(rule.getClass().getSimpleName());
+    for (final String alias : rule.aliases()) {
+      enabled &= Config.get().isRuleEnabled(alias);
+    }
+    if (!enabled) {
```

Comment:
I know that you did not added this but...what is that ??

Reply:
I think there was an SLF4J log, which is removed automatically